operator/controllers/seldondeployment_controller.go

```diff
@@ -743,6 +760,13 @@ func createDeploymentWithoutEngine(depName string, seldonId string, seldonPodSpe
 		},
 	}
 
+	if deploy.Spec.Template.Annotations == nil {
+		deploy.Spec.Template.Annotations = map[string]string{}
+	}
+	// Add prometheus annotations
+	deploy.Spec.Template.Annotations["prometheus.io/path"] = getPrometheusPath(mlDep)
+	deploy.Spec.Template.Annotations["prometheus.io/scrape"] = "true"
```

Comment:
do you need to add `prometheus.io/port` as well ?

Reply:
No in the new approach. We will now be using named ports in order to scrape multiple endpoints in one POD. You can see more discussion under the issue https://github.com/SeldonIO/seldon-core/issues/1476

Another reference: https://stackoverflow.com/questions/43121132/prometheus-with-multi-container-pod-on-kubernetes/46825817#46825817